src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -738,6 +739,22 @@ WhereClauseList(FromExpr *fromExpr)
 
 
 /*
+ * OuterJoinClauseList walks over the FROM expression in the query tree, and builds
+ * a list of outer join clauses from the expression tree.
+ */
+List *
+OuterJoinClauseList(FromExpr *fromExpr)
```

Comment:
> Perhaps we could merge two such that WhereClauseList returns join expressions for outer joins too. However, clauses that are used for filtering inside ON() clause should not be reported as it changes post-join filtering semantics.

That was one of the ideas I've considered. However, we can't pass a new variable to ExtractFromExpressionWalker to signal if we want outer joins conditions or not because the way expression_tree_walker works.

Do you have any specific suggestions?